src/schema/v2/me/myCollection.ts

```diff
@@ -121,29 +127,54 @@ export const MyCollection: GraphQLFieldConfig<any, ResolverContext> = {
     // @ts-expect-error FIXME: Make `page` is an optional param of `gravityOptions`
     delete gravityOptions.page
 
-    return collectionArtworksLoader("my-collection", gravityOptions)
-      .then(({ body, headers }) => {
-        return connectionFromArraySlice(body, options, {
-          arrayLength: parseInt(headers["x-total-count"] || "0", 10),
-          sliceStart: gravityOptions.offset,
-        })
-      })
-      .catch((error) => {
-        console.error("[schema/v2/me/my_collection] Error:", error)
-
-        if (error.message == "Collection Not Found") {
-          // For some users with no items, Gravity produces an error of
-          // "Collection Not Found". This can cause the Gravity endpoint to
-          // produce a 404, so we will intercept the error and return an empty
-          // list instead.
-          return connectionFromArray([], options)
-        } else {
-          throw error
-        }
+    try {
+      const { body: artworks, headers } = await collectionArtworksLoader(
+        "my-collection",
+        gravityOptions
+      )
+
+      const artworksWithSubmissions = await enrichArtworks(
+        artworks,
+        submissionsLoader,
+        info
+      )
+
+      return connectionFromArraySlice(artworksWithSubmissions, options, {
+        arrayLength: parseInt(headers["x-total-count"] || "0", 10),
+        sliceStart: gravityOptions.offset,
       })
+    } catch (error) {
+      console.error("[schema/v2/me/my_collection] Error:", error)
+
+      if (error.message == "Collection Not Found") {
+        // For some users with no items, Gravity produces an error of
+        // "Collection Not Found". This can cause the Gravity endpoint to
+        // produce a 404, so we will intercept the error and return an empty
+        // list instead.
+        return connectionFromArray([], options)
+      } else {
+        throw error
+      }
+    }
   },
 }
 
+const enrichArtworks = async (
+  artworks: any,
+  submissionsLoader: StaticPathLoader<any>,
+  _info: GraphQLResolveInfo
+) => {
+  const submissions = await submissionsLoader({ size: 1000 })
```

Comment:
This is temporary. SWA team will be working on enabling querying submissions via graphql and it will be possible to pass submission_ids taken from `artworks` . After that we can query only submission we need, not all user's submissions